### PR TITLE
Save task results in psql instead of redis

### PR DESCRIPTION
**Pull request by lbarcziova**
- Celery saves the task result data in psql instead of redis
- We create own table with the results (without Celery information, e.g.traceback) - API uses this table

### CODE REVIEW

File: packit_service/celerizer.py [hunk 1]
```diff
@@ -41,9 +42,10 @@ def celery_app(self):
             redis_url = "redis://{host}:{port}/{db}".format(
                 host=redis_host, port=redis_port, db=redis_db
             )
+            postgres_url = f"db+{get_pg_url()}"
 
             # http://docs.celeryproject.org/en/latest/reference/celery.html#celery.Celery
-            self._celery_app = Celery(backend=redis_url, broker=redis_url)
+            self._celery_app = Celery(backend=postgres_url, broker=redis_url)
```
Comment (TomasTomecek) [inline packit_service/celerizer.py]: wow, it's really that simple? :O no need to change backend type?

File: packit_service/service/api/tasks.py [hunk 2]
```diff
@@ -67,6 +68,14 @@ def get(self):
                     data["result"]["event"] = Event.ts2str(event)
                 tasks.append(data)
 
+        for task in TaskResult.get_all():
+            data = task.to_dict()
```
Comment (TomasTomecek) [inline packit_service/service/api/tasks.py]: wow, I didn't know that SA has such method, neat!

File: packit_service/models.py [hunk 3]
```diff
@@ -356,6 +357,45 @@ def __str__(self):
...
+                session.add(task_result)
+                return task_result
+            else:
+                task_result = cls(task_id)
+                task_result.status = status
+                task_result.result = result
+                task_result.traceback = traceback
+                task_result.date_done = date_done
+                session.add(task_result)
+                return task_result
```
Comment (lachmanfrantisek) [inline packit_service/models.py]: I know it's still WIP, but small improvement:

```suggestion
            if task_result is None:
                task_result = cls(task_id)
            task_result.status = status
            task_result.result = result
            task_result.traceback = traceback
            task_result.date_done = date_done
            session.add(task_result)
            return task_result
```

File: packit_service/service/api/tasks.py (hunk 2 above)
Comment (lbarcziova) [reply packit_service/service/api/tasks.py]: actually I think it is a method from celery task model

File: packit_service/celerizer.py [hunk 4]
```diff
@@ -41,9 +42,10 @@ def celery_app(self):
             redis_url = "redis://{host}:{port}/{db}".format(
                 host=redis_host, port=redis_port, db=redis_db
             )
+            postgres_url = f"db+{get_pg_url()}"
```
Comment (TomasTomecek) [inline packit_service/celerizer.py]: could you please add a comment explaining what the db+ prefix stands for?

File: packit_service/models.py [hunk 5]
```diff
@@ -356,6 +353,38 @@ def __str__(self):
         return self.__repr__()
 
 
+class TaskResult(Task):
```
Comment (TomasTomecek) [inline packit_service/models.py]: we should inherit from Base actually and define the fields ourselves

I recall that @jpopelka did not like this in redis that we would store the data twice, but on the other hand: one copy is handled by celery and the other one by us to use - alternatively we could write code which would retrieve data from the celery tables, not sure if we want that

my preferred solution is to have two copies and full control of the data, rather then rely on celery

Comment (jpopelka) [reply packit_service/models.py]: Sorry guys, I have no idea why we need this and have no connection to even read the docs so feel free to do whatever you like. If it's just about having some redundant data, well, ok, disk space is cheaper than memory (redis), so, go ahead.

Comment (TomasTomecek) [reply packit_service/models.py]: Oki, we can fight about this on the next architecture meeting :))

File: packit_service/celerizer.py (hunk 4 above)
Comment (lbarcziova) [reply packit_service/celerizer.py]: added

**Pull request merged**